tools/isobuild/import-scanner.js

```diff
@@ -362,53 +362,56 @@ export default class ImportScanner {
     return this;
   }
 
-  addNodeModules(identifiers) {
-    assert.ok(identifiers);
-    assert.ok(typeof identifiers === "object");
-    assert.ok(! Array.isArray(identifiers));
+  scanMissingModules(missingModules) {
+    assert.ok(missingModules);
+    assert.ok(typeof missingModules === "object");
+    assert.ok(! Array.isArray(missingModules));
 
     const newlyMissing = Object.create(null);
     const newlyAdded = Object.create(null);
 
-    if (! isEmpty(identifiers)) {
-      const previousAllMissingNodeModules = this.allMissingNodeModules;
-      this.allMissingNodeModules = newlyMissing;
+    if (! isEmpty(missingModules)) {
+      const previousAllMissingModules = this.allMissingModules;
+      this.allMissingModules = newlyMissing;
 
-      try {
-        this._scanFile({
+      Object.keys(missingModules).forEach(id => {
+        // TODO Call this._scanFile fewer times in case missingModules[id]
+        // is a long and redundant list?
+        missingModules[id].forEach(importInfo => this._scanFile({
           sourcePath: "fake.js",
           // It's important that the fake.js file itself never gets
           // scanned or bundled. See the _scanFile and getOutputFiles
           // methods for logic that deals with file.imported values.
           imported: "fake",
           lazy: true,
           // By specifying the .deps property of this fake file ahead of
-          // time, we can avoid calling findImportedModuleIdentifiers in the
-          // _scanFile method.
-          deps: identifiers,
-        });
+          // time, we can avoid calling findImportedModuleIdentifiers in
+          // the _scanFile method.
+          deps: { [id]: importInfo }
+        }));
+      });
 
-      } finally {
-        this.allMissingNodeModules = previousAllMissingNodeModules;
+      this.allMissingModules = previousAllMissingModules;
 
-        each(identifiers, (info, id) => {
-          if (! has(newlyMissing, id)) {
-            newlyAdded[id] = info;
-          }
-        });
+      Object.keys(missingModules).forEach(id => {
+        if (! has(newlyMissing, id)) {
+          newlyAdded[id] = missingModules[id];
```

Comment:
We don't need to be careful to use `ImportScanner.mergeMissing` here because this is the first time `newlyAdded[id]` has been assigned.